src/plans.js

```diff
@@ -364,11 +364,12 @@ function fetchNextJob (schema) {
             WHEN singletonKey IS NOT NULL
               AND singletonKey LIKE '${SINGLETON_QUEUE_KEY_ESCAPED}%'
               THEN NOT EXISTS (
-                SELECT *
+                SELECT 1
```

Comment:
@timgit made your requested changes, but…

I was still a little curious about load performance. I think my initial testing setup was too contrived (too many channels, not enough jobs per channel).

So I retested with 750 channels and 10,000 jobs queued up per channel. Those channels were broken into 3 groups: (A) 250 channels with jobs not using singleton keys, (B) 250 using singleton keys, and (C) 250 using singleton queue keys.

Operating assumptions for before and after my changes: It is reasonable that there would be an actual fetch performance hit on C channels (we’re doing more work), but for A or B channels we should expect basically the same performance.

Long story short, I can only satisfy the above when the JIT feature has been disabled. When JIT was on, fetch queries against A or B channels were up to 40x longer.

In general for a crafted, simple query like fetch, I think JIT can only hurt us. So ideally we’d just turn it off locally a la:
```sql
BEGIN;
SET LOCAL jit = OFF;
${theFetchQuery};
COMMIT;
```
But currently the query framework is using prepared statements, which do not support multiple commands.

So I think the options are:
1. “Prepare” queries ourselves before we ship them off. I don’t think we’re worried about sql injection, tho perhaps there's a small performance benefit
2. Use the [transaction capabilities](https://node-postgres.com/features/transactions#asyncawait) of the node pg library. The overhead here might be greater than any losses in no. 1 tho
3. Ship as is, but specifically caveat/flag the JIT issue. JIT is enabled by default on postgres >= 12

Thoughts? Personally I'd lean towards no. 1

Reply:
Thanks for the testing on this!  After doing a bit of reading on JIT, I'm convinced it's not worth leaving this decision to the query planner for a FOR UPDATE SKIP LOCKED query.  I agree with you on option 1.

A couple things that this change would require
1. Change the minimum supported version of pg to 11 in order to use `set local jit=off`
2. In order to do our own sanitation of the queue name, I'm imagining we'd write a regex of allowed characters that the API could check before allowing the job fetch.  Since bulk loading via `insert()` and direct table access is being used, it's possible that users would insert non-compliant queue names that are unreachable via the fetch and work API. In my opinion this is an acceptable caveat, but still discouraging. I think we should at least emit a warning event during maintenance if a non-compliant name is detected.  The alternative is to add something like a BEFORE trigger on the job table that enforces the same regex expression so we can block them on insert, which might be a better option to keep things consistent.

Reply:
Ok, pushed up some changes:

>1. Change the minimum supported version of pg to 11 in order to use set local jit=off

Done. Tho if preferable, this is technically only required if they're using the `enforceSingletonQueueActiveLimit` flag

>2. In order to do our own sanitation of the queue name, I'm imagining we'd write a regex of allowed characters that the API could check...

I was planning/hoping to be a little more surgical here and only change the query paradigm for fetch--i.e., not for publish/insert/etc--and only when the `enforceSingletonQueueActiveLimit` flag is on. Also wanted to avoid breaking the API contract if they're using a custom `executeSql` function via `options.db`. _(NOTE: If you want I could turn off JIT even when the flag is off, but in my testing I never saw JIT utilized for the current/unmodified fetch query)_

For sanitizing, I _think_ dollar-sign quoting the queue name input is sufficient (`batchSize` is already validated).

Let me know if I'm missing something